extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSource.java

```diff
@@ -116,8 +118,12 @@ private List<S3Object> fetchPrefixedS3Objects() {
         return s3Objects;
     }
 
+    private Collection<S3Object> filterOutFolderFile(List<S3Object> contents) {
+        return contents.stream().filter(object -> !object.key().endsWith("/")).collect(Collectors.toSet());
```

Comment:
the filter could be extracted in a field/constant like:
```
Predicate<S3Object> isFile = object -> !object.key.endsWith("/");
```

and this method could be inlined